Add customizable timestamp (#2788)
microsoft/BotFramework-WebChat: HOOKS.md

  }) => React.Element
): ({
  activity: Activity,
  nextVisibleActivity: Activity
}) => React.Element
```

This function is for rendering an activity and its attachments inside a React element. Because of the parent-child relationship, the caller will need to pass a render function in order for the attachment to create a render function for the activity. When rendering the activity, the caller will need to pass `activity` and `nextVisibleActivity`. This function is a composition of `activityRendererMiddleware`, which is passed as a prop.

> Because not all activities are rendered, for example, event activities are not rendered. The `nextVisibleActivity` is the pointer to next visible activity and is intended for activity status renderer on grouping timestamps for adjacent activities.

Reviewer: ```suggestion
Note that not all activities are rendered, e.g. the event activity. Because of this, those activities will not be rendered. The `nextVisibleActivity` is the pointer to the next visible activity and is intended for the activity status renderer on grouping timestamps for adjacent activities.
```
Add link to ACTIVITIES.md

### New in 4.8.0

Previously, we use `timestampClassName` to control if the activity should show timestamp or not. The `timestampClassName` should be add as a `class` attribute the DOM element which contains the timestamp.

> Today, we pass `activity` and `nextVisibleActivity` to the middleware, so the `activityRendererMiddleware` make the decision about timestamp visibility. For example, developers can group timestamp based on activity type.

Reviewer: We'll need to add this to the changelog. Please leave a note in R8 pre-release checklist

Author: Added to #2856.

## `useRenderActivityStatus`

```js
useRenderActivityStatus(): ({
  activity: Activity,
  nextVisibleActivity: Activity
}) => React.Element
```

> This function is for rendering status of an activity. The caller will need to pass `activity` and `nextVisibleActivity` as parameters. This function is a composition of `activityStatusRendererMiddleware`, which is passed as a prop.

Reviewer: ```suggestion
This function is for rendering the status of an activity. The caller will need to pass `activity` and `nextVisibleActivity` as parameters. This function is a composition of `activityStatusRendererMiddleware`, which is passed as a prop.
```

When called, this function will send a `postBack` activity to the bot.

## `useSendTimeoutForActivity`

```js
useSendTimeoutForActivity(): (activity: Activity) => number
```

> When called, this function will return a functino to evaluate the timeout (in milliseconds) for sending a specific activity.

Reviewer: ```suggestion
When called, this function will return a function to evaluate the timeout (in milliseconds) for sending a specific activity.
```
